database/gormdb_submission.go

```diff
@@ -80,8 +80,17 @@ func (db *GormDB) GetSubmission(query *pb.Submission) (*pb.Submission, error) {
 	return &submission, nil
 }
 
+// GetSubmissions returns all submissions matching the query.
+func (db *GormDB) GetSubmissions(query *pb.Submission) ([]*pb.Submission, error) {
```

Comment:
Do we need both these methods?
For this method, do you need all submissions, or just the latest?
From its point of use in `web/assignments.go:156`, it looks like you have access to the `courseID` in the `pb.Assignment` object, so you could use the same signature as `GetSubmissionsForCourse`.

Reply:
`GetSubmissionsForCourse` is twice as slow, but I guess it is not that important here. Removed the new method.

Reply:
`GetSubmissionsForCourse` will return one submission per each course assignment, and here we need all submissions for a single assignment, so these methods are not really interchangeable. Now that you changed the `GetSubmissionsByCourse` to be `GetSubmissions` there is a conflict here. What do you think I should name the new method than? It can be used to fetch multiple submissions by any parameter in the query, so, while i name it `GetSubmissionsByAssignment` for now, it is not really descriptive.

Reply:
My issue was not the name of the method, it was whether or not they were both needed. Looking at the code, the following naming feels more natural to me: 

```text
Current                      --> New
GetSubmissions()             --> GetLastSubmissions()  OR  GetLastSubmissionForEachAssignment()
GetSubmissionsByAssignment() --> GetSubmissions()
```

Reply:
To clarify, I'm okay with the two methods, since they provide different functionality.

Reply:
Renamed.